File: docs/vsto/codesnippet/VisualBasic/trin_contentcontrolreference/PlainText.vb
Partial Public NotInheritable Class ThisDocument

    ' PlainTextContentControl constructor #1: create a PlainTextContentControl at the current selection.
    '<Snippet600>
    Dim plainTextControl1 As Microsoft.Office.Tools.Word.PlainTextContentControl

    Private Sub AddPlainTextControlAtSelection()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        Me.Paragraphs(1).Range.Select()
        plainTextControl1 = Me.Controls.AddPlainTextContentControl("plainTextControl1")
        plainTextControl1.PlaceholderText = "Enter your first name"
    End Sub
    '</Snippet600>

    ' PlainTextContentControl constructor #2: create a PlainTextContentControl at a specified range.
    '<Snippet601>
    Dim plainTextControl2 As Microsoft.Office.Tools.Word.PlainTextContentControl

    Private Sub AddPlainTextControlAtRange()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        plainTextControl2 = Me.Controls.AddPlainTextContentControl(Me.Paragraphs(1).Range, "plainTextControl2")
        plainTextControl2.PlaceholderText = "Enter your first name"
    End Sub
    '</Snippet601>

    ' PlainTextContentControl constructor #3: create a PlainTextContentControl for every
    ' native plain text control in the document.
    '<Snippet602>
    Private plainTextControls As New System.Collections.Generic.List _
        (Of Microsoft.Office.Tools.Word.PlainTextContentControl)

    Private Sub CreatePlainTextControlsFromNativeControls()
        If Me.ContentControls.Count <= 0 Then
            Return
        End If

        Dim count As Integer = 0
        For Each nativeControl As Word.ContentControl In Me.ContentControls
            If nativeControl.Type = Word.WdContentControlType.wdContentControlText Then
                count += 1
                Dim tempControl As Microsoft.Office.Tools.Word.PlainTextContentControl = _
                    Me.Controls.AddPlainTextContentControl(nativeControl, _
                    "VSTOPlainTextContentControl" + count.ToString())
                plainTextControls.Add(tempControl)
            End If
        Next nativeControl
    End Sub
    '</Snippet602>

    ' PlainTextContentControl constructor #3: create a 
    ' PlainTextContentControl for every native plain text control added to the document.
    '<Snippet603>
    Private Sub ThisDocument_PlainTextContentControlAfterAdd(ByVal NewContentControl As Word.ContentControl, _
        ByVal InUndoRedo As Boolean) Handles Me.ContentControlAfterAdd

        If NewContentControl.Type = Word.WdContentControlType.wdContentControlText Then
            Me.Controls.AddPlainTextContentControl(NewContentControl, _
                "PlainTextControl" + NewContentControl.ID)
        End If
    End Sub
    '</Snippet603>

End Class

File: docs/vsto/codesnippet/CSharp/trin_wordcontentcontrolreference/PlainText.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word = Microsoft.Office.Interop.Word;
using Office = Microsoft.Office.Core;

namespace Trin_ContentControlReference
{
    public partial class ThisDocument
    {

        // PlainTextContentControl constructor #1: create a PlainTextContentControl at the current selection.
        //<Snippet600>
        private Microsoft.Office.Tools.Word.PlainTextContentControl textControl1;

        private void AddTextControlAtSelection()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();
            this.Paragraphs[1].Range.Select();

            textControl1 = this.Controls.AddPlainTextContentControl("textControl1");
            textControl1.PlaceholderText = "Enter your first name";
        }
        //</Snippet600>

        // PlainTextContentControl constructor #2: create a PlainTextContentControl at a specified range.
        //<Snippet601>
        private Microsoft.Office.Tools.Word.PlainTextContentControl textControl2;

        private void AddTextControlAtRange()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();

            textControl2 = this.Controls.AddPlainTextContentControl(this.Paragraphs[1].Range,
                "textControl2");
            textControl2.PlaceholderText = "Enter your first name";
        }
        //</Snippet601>

        // PlainTextContentControl constructor #3: create a PlainTextContentControl for every
        // native plain text control in the document.
        //<Snippet602>
        private System.Collections.Generic.List<Microsoft.Office.Tools.Word.PlainTextContentControl> plainTextControls;

        private void CreateTextControlsFromNativeControls()
        {
            if (this.ContentControls.Count <= 0)
                return;

            plainTextControls = new System.Collections.Generic.List
                <Microsoft.Office.Tools.Word.PlainTextContentControl>();
            int count = 0;

            foreach (Word.ContentControl nativeControl in this.ContentControls)
            {
                if (nativeControl.Type == Word.WdContentControlType.wdContentControlText)
                {
                    count++;
                    Microsoft.Office.Tools.Word.PlainTextContentControl tempControl =
                        this.Controls.AddPlainTextContentControl(nativeControl,
                        "VSTOPlainTextContentControl" + count.ToString());
                    plainTextControls.Add(tempControl);
                }
            }
        }
        //</Snippet602>

        // PlainTextContentControl constructor #3: create a PlainTextContentControl for every
        // native plain text control that is added to the document at run time.
        //<Snippet603>
        void ThisDocument_PlainTextContentControlAfterAdd(Word.ContentControl NewContentControl, bool InUndoRedo)
        {
            if (NewContentControl.Type == Word.WdContentControlType.wdContentControlText)
            {
                this.Controls.AddPlainTextContentControl(NewContentControl,
                    "PlainTextControl" + NewContentControl.ID);
            }
        }
        //</Snippet603>

    }
}
